From 455020ec38e08e40d7fad7043f62bf95edd1280f Mon Sep 17 00:00:00 2001
From: Bharath Paidupati <bpaidupati@maxlinear.com>
Date: Wed, 24 Jul 2024 16:31:31 +0530
Subject: [PATCH] WLANRTSYS-64411 Unconnected STA Enhancement to support 320MHz

Since 11BE supports 320MHz bandwidth and trying to test the feature with
320MHz. It was throwing error saying invalid bandwidth provided. Now we
have added the support for 320MHz.

Fixup to: WLANRTSYS-52730: AP: Beerock implementation
---
 hostapd/ctrl_iface.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 700c0b412..8acaa8f36 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -5951,6 +5951,9 @@ static int hostapd_ctrl_iface_unconnected_sta(struct hostapd_data *hapd, const c
 		case 160:
 			req.bandwidth = NL80211_CHAN_WIDTH_160;
 			break;
+		case 320:
+			req.bandwidth = NL80211_CHAN_WIDTH_320;
+			break;
 		default:
 			wpa_printf(MSG_ERROR, "unconnected_sta: invalid bandwidth provided");
 			goto err;
-- 
2.43.0

